content/post/kubedb-v2024.06.04/index.md

```diff
@@ -0,0 +1,788 @@
+  databaseRef:
+    name: pgpool
+  verticalScaling:
+    node:
+      resources:
+        requests:
+          memory: "1200Mi"
+          cpu: "1"
+        limits:
+          memory: "1800Mi"
+          cpu: "2"
+```
+
+#### Reconfigure
+Reconfigure allows you to reconfigure Pgpool with new configuration via `spec.configuration.configSecret` or `spec.configuration.applyConfig` can be used if you want to apply changes or add some new configuration in addition to currently used configuration. Also, you can remove currently used configuration and use the default configuration by using `spec.configuration.removeCustomConfig`. An example YAML is provided below:
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: PgpoolOpsRequest
+metadata:
+  name: pgpool-reconfigure
+  namespace: demo
+spec:
+  type: Reconfigure
+  databaseRef:
+    name: pgpool
+  configuration:
+    applyConfig:
+      pgpool.conf: |-
+        memory_cache_enabled = on
+```
+
+### AutoScaler
+In this release, we are introducing PgpoolAutoscaler, a Kubernetes Custom Resource Definition (CRD) that supports auto scaling for Pgpool. This allows you to configure auto scaling for Pgpool based on cpu, memory and nodeTopology. An example YAML is provided below:
+```yaml
+apiVersion: autoscaling.kubedb.com/v1alpha1
+kind: PgpoolAutoscaler
+metadata:
+  name: pgpool-autoscaler
+  namespace: demo
+spec:
+  databaseRef:
+    name: pgpool
+  compute:
+    pgpool:
+      trigger: "On"
+      podLifeTimeThreshold: 5m
+      minAllowed:
+        cpu: 3
+        memory: 2Gi
+      maxAllowed:
+        cpu: 5
+        memory: 5Gi
+      resourceDiffPercentage: 5
+      controlledResources: ["cpu", "memory"]
+```
+### Custom configuration via config secret
+Now we can create a secret with `pgpool.conf` as the key and  refer this secret to Pgpool for use. KubeDB operator will validate this secret and merge with default configuration and use it for Pgpool. Just give the reference for the config secret to `spec.configSecret` and the KubeDB operator will do the rest for you. We can also give custom configuration via `spec.initConfig` as previous versions.
+### Pod Disruption Budget (PDB)
+We are now automatically creating a Pod Disruption Budgets (PDB) for Pgpool upon creating PetSet for it. A PDB helps ensure the availability of Pgpool by limiting the number of pods that can be down simultaneously due to voluntary disruptions (e.g., maintenance or upgrades).
+### New service port for pcp user
+Now you can use pcp users through the primary service to do administrative level tasks for Pgpool. By default the port is `9595`.
+
+
+## Postgres
+In this release, we have added a PostgreSQL extension for the Apache AGE Graph Database in PostgreSQL version 15. This extension will be supported on Linux Alpine and Debian-based PostgreSQL images. Additionally, Remote Replica will be supported for PostgreSQL major versions 13 and 14. As a reminder, we already have support for major versions 15 and 16.
```

Comment:
There is two different point. Make them seperate.

Reply:
updated